File: api/src/main/java/net/blueshell/api/service/email/EmailDeliveryService.java
package net.blueshell.api.service.email;

import jakarta.mail.MessagingException;
import jakarta.mail.internet.MimeMessage;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.ClassPathResource;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Service;

import java.io.UnsupportedEncodingException;

@Service
@Slf4j
public class EmailDeliveryService {

    @Autowired
    private JavaMailSender mailSender;

    /**
     * Sends an HTML email (synchronous). Jobs wrap retries/async.
     */
    public void sendHtmlEmail(String toEmail,
                              String subject,
                              String htmlContent,
                              String senderName,
                              String senderAddress) {
        try {
            MimeMessage message = mailSender.createMimeMessage();
            MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");

            helper.setFrom(senderAddress, senderName);
            helper.setTo(toEmail);
            helper.setSubject(subject);
            helper.setText(htmlContent, true);

            // keep your inline assets
            helper.addInline("logo", new ClassPathResource("templates/assets/BSLOGO.png"), "image/png");
            helper.addInline("bg", new ClassPathResource("templates/assets/BackdropBlack.jpg"), "image/jpeg");

            mailSender.send(message);
            log.info("Sent email to {} with subject {}", toEmail, subject);
        } catch (MessagingException | UnsupportedEncodingException e) {
            log.error("Failed to send email to {} with subject {}: {}", toEmail, subject, e.getMessage(), e);
            throw new RuntimeException("Failed to send email", e);
        }
    }
}

File: api/src/main/kotlin/net/blueshell/api/platform/integration/email/service/EmailDeliveryService.kt
package net.blueshell.api.platform.integration.email.service

import net.blueshell.api.platform.integration.email.EmailTransportClient
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class EmailDeliveryService(private val emailClient: EmailTransportClient) {

    /**
     * Sends an HTML email via SMTP (blueshell.utwente.nl relay).
     * Returns the Message-ID for outbox correlation.
     * Jobs wrap retries/async.
     */
    fun sendHtmlEmail(
        toEmail: String,
        toName: String,
        subject: String,
        htmlContent: String,
        senderName: String,
        senderAddress: String,
        replyToAddress: String,
    ): String {
        try {
            val messageId = emailClient.send(toEmail, toName, subject, htmlContent, senderName, senderAddress, replyToAddress)
            log.info("Sent email to {} from {} with subject {}", toEmail, senderAddress, subject)
            return messageId
        } catch (e: Exception) {
            log.error("Failed to send email to {} with subject {}: {}", toEmail, subject, e.message, e)
            throw RuntimeException("Failed to send email", e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(EmailDeliveryService::class.java)
    }
}
